ihmc-perception/src/main/java/us/ihmc/perception/gpuHeightMap/RapidHeightMapExtractor.java

```diff
@@ -0,0 +1,549 @@
+import org.bytedeco.javacpp.FloatPointer;
+import org.bytedeco.javacpp.Pointer;
+import org.bytedeco.opencv.global.opencv_core;
+import org.bytedeco.opencv.opencv_core.GpuMat;
+import org.bytedeco.opencv.opencv_core.Scalar;
+import us.ihmc.euclid.axisAngle.AxisAngle;
+import us.ihmc.euclid.matrix.interfaces.RotationMatrixBasics;
+import us.ihmc.euclid.transform.RigidBodyTransform;
+import us.ihmc.euclid.transform.interfaces.RigidBodyTransformReadOnly;
+import us.ihmc.euclid.tuple3D.Point3D;
+import us.ihmc.euclid.tuple3D.interfaces.Vector3DBasics;
+import us.ihmc.perception.camera.CameraIntrinsics;
+import us.ihmc.perception.cuda.CUDAKernel;
+import us.ihmc.perception.cuda.CUDAProgram;
+import us.ihmc.perception.cuda.CUDAStreamManager;
+import us.ihmc.perception.cuda.CUDATools;
+import us.ihmc.sensorProcessing.heightMap.HeightMapParameters;
+import us.ihmc.sensorProcessing.heightMap.HeightMapTools;
+
+import java.net.URL;
+
+import static org.bytedeco.cuda.global.cudart.*;
+
+public class RapidHeightMapExtractor
+{
+   private static final boolean PRINT_TIMING_FOR_KERNELS = false;
+   static final int BLOCK_SIZE_XY = 32;
+   private static final int MODE = 1; // 0 -> Ouster, 1 -> Realsense
+
+   private final HeightMapParameters heightMapParameters;
+   private final CUstream_st stream;
+   private final CUDAProgram heightMapProgram;
+   private final dim3 blockSize;
+
+   // These are the mats required to extract the depth data
+   private final GpuMat localMeanMap;
+   private final GpuMat localVarianceMap;
+   private final GpuMat localMotionVarianceMap;
+   private final GpuMat localSampleCountMap;
+
+   // These are the mats required to keep a global map
+   private final GpuMat globalMeanMap;
+   private final GpuMat globalVarianceMap;
+   private final GpuMat previousGlobalMeanMap;
+   private final GpuMat previousGlobalVarianceMap;
+   private final GpuMat terrainCroppedHeightMap;
+   private final GpuMat scaledHeightMap;
+   private final GpuMat emptyGlobalHeightMap;
+
+   private final CUDAKernel updateKernel;
+   private final CUDAKernel translateKernel;
+   private final CUDAKernel registerKernel;
+   private final CUDAKernel terrainCroppingKernel;
+   private final CUDAKernel scalingKernel;
+   private final CUDAKernel planOffsetKernel;
+   private final CUDAKernel emptyRegisterKernel;
+
+   private final float[] worldToGroundTransformArray = new float[16];
+   private final float[] groundToSensorTransformArray = new float[16];
+   private final float[] sensorToGroundTransformArray = new float[16];
+
+   private final FloatPointer groundToSensorTransformHostPointer;
+   private final FloatPointer groundToSensorTransformDevicePointer;
+   private final FloatPointer sensorToGroundTransformHostPointer;
+   private final FloatPointer sensorToGroundTransformDevicePointer;
+   private final FloatPointer zUpCameraToWorldAlignedGroundHostPointer;
+   private final FloatPointer ZUpCameraToWorldAlignedGroundDevicePointer;
```

Comment:
Don't apply this through GitHub, but start field name with lower case
```suggestion
   private final FloatPointer zUpCameraToWorldAlignedGroundDevicePointer;
```